### PR TITLE
[RW-4506][risk=no] keep elastic at pace with CB changes

### PR DESCRIPTION
**Pull request by freemabd**
Not sure if Elastic will see the light of day in AoU, but I'm trying to keep it up to date with all CB changes. CB changes(feature flagged):

1. CB can now search by Current Age, Age at Consent and Age at CDR.
image

2. CB graphs will allow you to refresh graphs based on Age Range(Current, Consent or CDR), Gender or Sex at Birth and Race.
image

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/elasticsearch/AggregationUtils.java [hunk 1]
```diff
@@ -10,50 +10,64 @@
...
-                        .order(BucketOrder.key(true))
-                        // This eliminates the race buckets with 0 counts. Without this param
-                        // elastic
-                        // returns all race buckets regardless of count.
-                        .minDocCount(1)));
+    boolean isGender = GenderOrSexType.GENDER.equals(genderOrSexType);
+    boolean isAgeAtConsent = AgeType.AGE_AT_CONSENT.equals(ageType);
+    // Added order to gender/sex and race buckets. Therefore the UI code can expect consistent
+    // results
+    // between BQ(sql ordered by gender/sex, race, age) and elastic
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/elasticsearch/AggregationUtils.java]: nit: fix newline

File: api/src/test/java/org/pmiops/workbench/elasticsearch/ElasticFiltersTest.java [hunk 2]
```diff
@@ -925,18 +927,50 @@ public void testAgeQuery() {
...
                     new SearchGroup()
-                        .addItemsItem(new SearchGroupItem().addSearchParametersItem(ethParam))));
+                        .addItemsItem(new SearchGroupItem().addSearchParametersItem(ageParam))));
     BoolQueryBuilder ageBuilder =
         QueryBuilders.boolQuery()
             .filter(QueryBuilders.termQuery("is_deceased", false))
             .filter(
                 QueryBuilders.rangeQuery("birth_datetime")
-                    .gt(left)
+                    .gte(left)
```
Comment (calbach) [inline api/src/test/java/org/pmiops/workbench/elasticsearch/ElasticFiltersTest.java]: Hm - did this change? Or was this a bug?

File: api/tools/src/main/java/org/pmiops/workbench/tools/elastic/ElasticSearchIndexer.java [hunk 3]
```diff
@@ -306,9 +306,11 @@ private void createIndex(CommandLine opts) throws IOException, InterruptedExcept
   }
 
   private String getPersonBigQuerySQL(String bqDataset, int inverseProb) throws IOException {
+    // Synth dataset doesn't contain sex_at_birth, so we use gender as a fix
```
Comment (calbach) [inline api/tools/src/main/java/org/pmiops/workbench/tools/elastic/ElasticSearchIndexer.java]: Owch. Will check for an update on synthetic data this week. We need to be able to rerun the latest curation pipeline at a bare minimum even if we can't easily make data improvements yet.

File: api/src/main/java/org/pmiops/workbench/elasticsearch/AggregationUtils.java (hunk 1 above)
Comment (freemabd) [reply api/src/main/java/org/pmiops/workbench/elasticsearch/AggregationUtils.java]: Done.

File: api/src/test/java/org/pmiops/workbench/elasticsearch/ElasticFiltersTest.java (hunk 2 above)
Comment (freemabd) [reply api/src/test/java/org/pmiops/workbench/elasticsearch/ElasticFiltersTest.java]: A bug. So participants that have birthdays today wouldn't be included in the left most range.

File: api/tools/src/main/java/org/pmiops/workbench/tools/elastic/ElasticSearchIndexer.java (hunk 3 above)
Comment (freemabd) [reply api/tools/src/main/java/org/pmiops/workbench/tools/elastic/ElasticSearchIndexer.java]: Agreed.

**Pull request merged**